packages/expo-polyfills/src/storage.ts

```diff
@@ -139,6 +141,56 @@ export class SyncSecureStorage implements Storage {
     this.setKeys();
   }
 
+  private async getValueForKey(key: string): Promise<string | null> {
+    let valStr = await SecureStore.getItemAsync(key);
+    if (!valStr) {
+      return valStr;
+    }
+    let chunk = 0;
+    while (valStr.endsWith('___continued___')) {
+      chunk++;
+      valStr = valStr.replace('___continued___', '');
+      const next = await SecureStore.getItemAsync(`${key}___chunk${chunk}___`);
+      if (!next) {
+        this.removeItem(key);
+        throw new OperationOutcomeError({
+          id: 'key-chunk-missing',
+          resourceType: 'OperationOutcome',
+          issue: [
+            {
+              severity: 'error',
+              code: 'exception',
+              details: {
+                text: 'Chunk missing for key',
+              },
+              diagnostics: `Chunk ${chunk} of key's value is missing for key '${key}'`,
+            },
+          ],
+        });
+      }
+      valStr += next;
+    }
+    return valStr;
+  }
+
+  private async setValueForKey(key: string, value: string): Promise<void> {
+    const chunks = splitStringIntoChunks(value, MAX_VALUE_LENGTH);
+    // Append ___continued___ if this isn't the only chunk
+    // Otherwise just put the full chunk in
+    const promises = [];
+    promises.push(SecureStore.setItemAsync(key, chunks.length === 1 ? chunks[0] : `${chunks[0]}___continued___`));
+    for (let i = 1; i < chunks.length; i++) {
+      promises.push(
+        SecureStore.setItemAsync(
+          `${key}___chunk${i}___`,
+          // If this isn't the last chunk append ___continued___ to it
+          i === chunks.length - 1 ? chunks[i] : `${chunks[i]}___continued___`
```

Comment:
If `chunks[i]` is 2000 characters long, will appending `__continued__` exceed the limit?

Reply:
The limit is 2048, so 2000 + 13 shouldn't exceed. Good question though